Add OTTL delete function (#44053)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

- `append(log.attributes["tags"], values = ["staging", "staging:east"])`
- `append(log.attributes["tags_copy"], log.attributes["tags"])`

### delete_index

> `delete(target, startIndex, Optional[endIndex])`

Reviewer: ```suggestion
`delete_index(target, startIndex, Optional[endIndex])`
```

> The `delete` function removes elements from a slice. It deletes elements in the range `[startIndex, endIndex)` (startIndex inclusive, endIndex exclusive). If `endIndex` is not provided, only the element at `target[startIndex]` is deleted. If `startIndex` equals `endIndex`, no changes are applied to the target, following the behavior of Go's `slices.Delete(s, i, j)` function.

Reviewer: ```suggestion
The `delete` function removes elements from a slice. It deletes elements from `startIndex` up to, but not including, `endIndex`. If `endIndex` is not provided, only the element at `target[startIndex]` is deleted. If `startIndex` equals `endIndex`, no changes are applied to the target.
```
